glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -195,9 +197,31 @@ open class GleanInternalAPI internal constructor () {
         branch: String,
         extra: Map<String, String>? = null
     ) {
-        Log.e(LOG_TAG, "setExperimentActive is a stub")
-        // TODO: 1552471 stub
-        // ExperimentsStorageEngine.setExperimentActive(experimentId, branch, extra)
+        if (!isInitialized()) {
+            Log.e(LOG_TAG, "Please call Glean.initialize() before using this API")
+            return
+        }
+
+        // The 'extra' map is sent over FFI as a pair of arrays, one containing
+        // the keys and the other containing the values.
+        var keys: StringArray? = null
+        var values: StringArray? = null
+        var numKeys = 0
+
+        extra?.let {
+            numKeys = extra.size
+            keys = StringArray(extra.keys.toTypedArray(), "utf-8")
+            values = StringArray(extra.values.toTypedArray(), "utf-8")
```

Comment:
As discussed in #161, technically according to the API, this could result in the keys and values being in a different order.  I think to be certain, we need to iterate over the key/value pairs.

Reply:
Thank you, fixed this.